library/std/src/sync/nonpoison/condvar.rs

```diff
@@ -115,16 +115,15 @@ impl Condvar {
     /// let mut started = lock.lock();
     /// // As long as the value inside the `Mutex<bool>` is `false`, we wait.
     /// while !*started {
-    ///     started = cvar.wait(started);
+    ///     cvar.wait(&mut started);
     /// }
     /// ```
     #[unstable(feature = "nonpoison_condvar", issue = "134645")]
-    pub fn wait<'a, T>(&self, guard: MutexGuard<'a, T>) -> MutexGuard<'a, T> {
+    pub fn wait<'a, T>(&self, guard: &mut MutexGuard<'a, T>) {
```

Comment:
```suggestion
    pub fn wait<T>(&self, guard: &mut MutexGuard<'_, T>) {
```
The explicit lifetime isn't needed any more.